### PR TITLE
Refactor `SpiceClient` to support connection to local spice runtime

### LINKED ISSUE #150: Add support for local Spice OSS runtime to spice.js

### PR DESCRIPTION
**Pull request by ewgenius**
closes #150 

**Partially breaking change**

The `SpiceClient` constructor has been modified to accept an optional object of `SpiceClientConfig` type instead of separate arguments

By default spice client uses connection to localhost (expecting to have local https://github.com/spiceai/spiceai runtime running)

```ts
const client = new SpiceClient();

// equals to 

const client = new SpiceClient({
  http_url: 'http://localhost:3000',
  flight_url: 'localhost:50051',
  flight_tls_enabled: false
});
```

For spice.ai cloud connection, use following configuration:

```ts
const client = new SpiceClient({
  api_key: '<spice.ai app api key>',
  http_url: 'https://data.spiceai.io',
  flight_url: 'flight.spiceai.io:443',
  flight_tls_enabled: true
});
```

Connecting with api_key as a first argument is still supported
```ts
const client = new SpiceClient('<spice.ai app api key>');
```

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -43,22 +44,37 @@ const arrow = grpc.loadPackageDefinition(packageDefinition).arrow as any;
...
 
-  public constructor(
-    apiKey: string,
-    http_url: string = 'https://data.spiceai.io',
-    flight_url: string = 'flight.spiceai.io:443'
-  ) {
-    this._apiKey = apiKey;
-    this._http_url = http_url;
-    this._flight_url = flight_url;
+  public constructor({
```
Comment (sgrebnov) [inline src/client.ts]: Nit: we can do the following to not break the most common use-case where API-KEY is provided as single string param (if we want to)
```
  public constructor(params: SpiceClientConfig | string ) {
    if (typeof params === 'string') {
      params = { api_key: params };
    }
```
// Above should also add default values for Spice cloud as well
 http_url: 'https://data.spiceai.io',
  flight_url: 'flight.spiceai.io:443',
  flight_ssl_enabled: false

File: test/scripts/setup-data-postgresql.sql [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  CURRENT_TIMESTAMP, 
+  CURRENT_DATE, 
+  ARRAY[1, 2], 
+  ARRAY[3, 4], 
+  ARRAY[5, 6], 
+  ARRAY[7.0, 8.0], 
+  ARRAY[9.0, 10.0], 
+  ARRAY['test1', 'test2'], 
+  ARRAY[true, false]
+), (
```
Comment (sgrebnov) [inline test/scripts/setup-data-postgresql.sql:84]: Like it 👍

**Review by sgrebnov — APPROVED** [review]

**Review by digadeesh — APPROVED** [review]

**Comment by digadeesh** [comment]
Note - we'll need to update the readme too.

File: src/client.ts (hunk 1 above)
Comment (ewgenius) [reply src/client.ts]: Thanks @sgrebnov! Updated

**Review by sgrebnov — APPROVED** [review]

**Review by digadeesh — APPROVED** [review]

**Pull request merged**